modules/cognito-user-pool/pool.tf

```diff
@@ -57,14 +57,20 @@ resource "aws_cognito_user_pool" "pool" {
   }
 
   dynamic "lambda_config" {
-    for_each = (var.lambda_config.kms_key_id != "" && var.lambda_config.custom_email_sender.lambda_arn != "" && var.lambda_config.custom_email_sender.lambda_version != "") ? [1] : []
+    for_each = length(keys(var.lambda_config)) != 0 ? [1] : []
 
     content {
-      kms_key_id = var.lambda_config.kms_key_id
-
-      custom_email_sender {
-        lambda_arn     = var.lambda_config.custom_email_sender.lambda_arn
-        lambda_version = var.lambda_config.custom_email_sender.lambda_version
+      kms_key_id                     = try(var.lambda_config.kms_key_id, null)
+      create_auth_challenge          = try(var.lambda_config.create_auth_challenge, null)
+      define_auth_challenge          = try(var.lambda_config.define_auth_challenge, null)
+      verify_auth_challenge_response = try(var.lambda_config.verify_auth_challenge_response, null)
+
+      dynamic "custom_email_sender" {
```

Comment:
this dynamic block will never created because for_each is always equal to [], won't it?

Reply:
fixed please check